internal/faketime/faketime.go

```diff
@@ -30,3 +30,36 @@ func AutoAdvance(t time.Time, dt time.Duration) func() time.Time {
 		return ret
 	}
 }
+
+// TimeAdvance allows controlling the passage of time. Intended to be used in
+// tests.
+type TimeAdvance struct {
+	mu sync.Mutex
+	t  time.Time
+}
```

Comment:
```suggestion
type TimeAdvance struct {
	base  time.Time
	int64 advance
}
```

Then
```go
func (t *TimeAdvance) NowFunc() func() time.Time {
	return func() time.Time {
		dt := time.Duration(atomic.LoadInt64((*int64)(&t.delta)))

		return t.base.Add(dt)
	}
}

func (t *TimeAdvance) Advance(dt time.Duration) time.Time {
	advance := atomic.AddInt64((*int64)(&t.delta), int64(t.delta))

	return t.base.Add(time.Duration(advance))
}
```

Reply:
oh, I meant time.Unix(0, atomic.AddInt64(...))

Reply:
but that's fine either way.